### PR TITLE
Chapter 4 & 5: Rails Flavored Ruby & Filling In The Layout

### PR DESCRIPTION
**Pull request by megatimranikram**

### CODE REVIEW

**Comment by megatimranikram** [comment]
ready sir

**Comment by megatimranikram** [comment]
ready

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -1,2 +1,6 @@
 module ApplicationHelper
-end
+  def full_title page_title
+    base_title = "Ruby on Rails Tutorial Sample App"
+    page_title.present? ? [page_title, base_title].join(" | ") : base_title
+  end
+end
```
Comment (ngocvt-0484) [inline app/helpers/application_helper.rb]: please add endline
No12
https://docs.google.com/document/d/104Csp4-vamVos5DEbi372nLBEfmqhj7h1ENYO8ebjCo/edit

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+class User < ApplicationRecord
+end
```
Comment (ngocvt-0484) [inline app/models/user.rb]: please add endline
No12
https://docs.google.com/document/d/104Csp4-vamVos5DEbi372nLBEfmqhj7h1ENYO8ebjCo/edit

File: config/routes.rb [hunk 3]
```diff
@@ -1,3 +1,5 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
-end
+  root "static_pages#home"
+
+  get "static_pages/help"
+end
```
Comment (ngocvt-0484) [inline config/routes.rb]: please add endline
No12
https://docs.google.com/document/d/104Csp4-vamVos5DEbi372nLBEfmqhj7h1ENYO8ebjCo/edit

**Comment by megatimranikram** [comment]
ready

File: app/models/user.rb (hunk 2 above)
Comment (ngocvt-0484) [reply app/models/user.rb]: not fix yet

Comment (ngocvt-0484) [inline app/models/user.rb]: please add endline
No12
https://docs.google.com/document/d/104Csp4-vamVos5DEbi372nLBEfmqhj7h1ENYO8ebjCo/edit

File: app/assets/stylesheets/custom.scss [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+ text-transform: uppercase;
+  letter-spacing: -1px;
+  padding-top: 9px;
+  font-weight: bold;
+}
+
+#logo:hover {
+  color: $white;
+  text-decoration: none;
+}
```
Comment (ngocvt-0484) [inline app/assets/stylesheets/custom.scss]: please add endline
No12
https://docs.google.com/document/d/104Csp4-vamVos5DEbi372nLBEfmqhj7h1ENYO8ebjCo/edit

File: app/assets/stylesheets/color.scss [hunk 5]
```diff
@@ -0,0 +1,3 @@
+$light-gray: #777;
+$gray-medium-light: #eaeaea;
+$white: #fff;
```
Comment (ngocvt-0484) [inline app/assets/stylesheets/color.scss]: please add endline
No12
https://docs.google.com/document/d/104Csp4-vamVos5DEbi372nLBEfmqhj7h1ENYO8ebjCo/edit

**Comment by ngocvt-0484** [comment]
please change 2 commits to 1 commit

**Comment by megatimranikram** [comment]
ready

File: app/assets/stylesheets/application.css [hunk 6]
```diff
@@ -13,3 +13,6 @@
  *= require_tree .
  *= require_self
  */
+
+ @import 'bootstrap-sprockets';
+ @import 'bootstrap';
```
Comment (chungpv-1008) [inline app/assets/stylesheets/application.css]: add endline for this file

File: config/application.rb [hunk 7]
```diff
@@ -10,4 +10,4 @@ class Application < Rails::Application
     config.i18n.available_locales = [:en, :ru]
     config.i18n.default_locale = :en
   end
-end
+end
```
Comment (chungpv-1008) [inline config/application.rb]: add endline for this file

File: config/locales/en.yml [hunk 8]
```diff
@@ -1,2 +1,2 @@
 en:
-  hello: "Hello world"
+  hello: "Hello world"
```
Comment (chungpv-1008) [inline config/locales/en.yml]: add endline for this file

**Comment by megatimranikram** [comment]
ready

File: app/assets/stylesheets/application.css (hunk 6 above)
Comment (megatimranikram) [reply app/assets/stylesheets/application.css]: done sir

File: config/application.rb (hunk 7 above)
Comment (megatimranikram) [reply config/application.rb]: done sir

File: config/locales/en.yml (hunk 8 above)
Comment (megatimranikram) [reply config/locales/en.yml]: done sir

**Comment by ngocvt-0484** [comment]
merged

**Pull request merged**